### ISSUE TITLE
option for bracket pair vertical line depending on indentation

### ISSUE THREAD

**Issue by Richhoef** [microsoft/vscode#136748]
The new Bracket Pairs Horizontal settings came together with the vertical line depending on the indentation. I would like this dependency to be an option, because I do not like it. My `#if` and `#endif` are situated at the start of the line, rendering all the vertical lines useless:
![image](https://user-images.githubusercontent.com/51912545/140924645-8700087c-94ae-45f7-9112-185373f6ca16.png)

**Comment by Richhoef** [comment]
@hediet, Why does this option need community votes. 
I am a bit afraid a lot of people do annoy this behaviour but won't go and file an issue/search and find this issue to upvote it. I would immagine this would be a very easy option to add. The desired behaviour was already happening in the October version of VSCode, it should just be switchable on/off with an option.

The not-indenting of `#if`/`#ifdef`/`#endif` is very common in C/C++, so for quite some people, the vertical bracket guides are rather useless right now. 

I already looked at the code myself, but I was not able to compile VSCode. I am not quite home in *.ts files, but I would imagine it is as easy as changing the second argument in the call to `getBracketPairsInRange()` in the file `bracketPairsImpl.ts` on line 100, depending on this new option, from `true` to `false` (and of course adding the option itself).

**Comment by hediet** [comment]
> I am a bit afraid a lot of people do annoy this behaviour but won't go and file an issue/search and find this issue to upvote it.

Fair point, but would a setting really help them? They would need to know about it.
We try to only add settings when really necessary.

I understand your use-case, but before adding yet another setting, I would love to know how many people actually run into this.

**Comment by Richhoef** [comment]
> Fair point, but would a setting really help them? They would need to know about it. We try to only add settings when really necessary.

That is also a fair point, but I do think there is a huge gap between searching for a setting in VSCode or going to github and searching/filing an issue. 

Of course I do not have numbers on how many people run into this, but since such defines are very common in C/C++ and C/C++ are quite popular programming languages (I do not know if it also common in other programming languages), I would imagine quite a lot of people would run into this and benefit from such a setting.
